src/transformers/utils/quantization_config.py

```diff
@@ -418,3 +423,51 @@ def post_init(self):
                     f"""dataset needs to be either a list of string or a value in
                     ['wikitext2','c4','c4-new','ptb','ptb-new'], but we found {self.dataset}"""
                 )
+        if self.bits == 4:
+            if self.disable_exllama is not None:
+                logger.warning(
+                    "Using `disable_exllama` is deprecated and will be removed in future version. Use `use_exllama` instead."
+                )
+                self.use_exllama = not self.disable_exllama
```

Comment:
No, we shouldn't do this. It's possible a user does the following: 

```
config = GPTQConfig.from_pretrained(checkpoint, use_exllama=True)
```

in which the checkpoint had `disable_exllama=True` in its saved config. This will lead to silent behaviour and unexpected behaviour. 

Instead, we should raise and error if `disable_exllama` and `use_exllama` are both `True`

Reply:
Good catch. I guess there is no perfect solution in how we should handle these args in case the arg is in its saved config . There are multiple problem with raising an error if `disable_exllama` and `use_exllama` are both `True`:
- if `disable_exllama` is in the saved config and both are True, it will error out. To solve this, we will have to set both `disable_exllama` and `use_exllama` to fix the error.
-  `disable_exllama` become useless as it does not have an effect on `use_exllama` -> problem for users who set 
`disable_exllama` = True 

What I propose is to keep `self.use_exllama = not self.disable_exllama` and add a better warning saying that we used the value of `disable_exllama`  instead of `use_exllama`, so that the BC is preserved. Then in 4.37, we will remove `disable_exllama`. 

PS: Most gptq model were not quantized using transformers quantizer, so they don't have `disable_exllama` in their saved config .

Reply:
We can't have the current behaviour at the moment because it fails forwards compatibility with the `use_exllama` argument. 

> I guess there is no perfect solution in how we should handle these args in case the arg is in its saved config . There are multiple problems with raising an error if disable_exllama and use_exllama are both True

As you point out, erroring out could cause issues so let's set them in the init. What I would suggest is something like this:

```
class GPTQConfig(QuantizationConfigMixing):
    def __init__(
        use_exllama: bool = None,
    	**kwargs
    ):

    disable_exllama = kwargs.pop("disable_exllama", None)

    if disable_exllama is None and use_exllama is None: 
        # New default behaviour
        use_exllama = True
    elif disable_exllama is not None and use_exllama is None:
        # Follow pattern of old config
        use_exllama = not disable_exllama
    elif disable_exllama is not None and use_exllama is not None:
        # Only happens if user explicitly passes in both arguments
        raise ValueError("Cannot specify both disable_exllama and use_exllama. Please use just use_exllama")
```

This way: 
* `use_exllama` defaults to `True` for new configs 
* `use_exllama` takes `not disable_exllama`'s for old configs
* Error is raised only if both are explicitly set

Reply:
Thanks for the suggestion ! That's indeed better.